### DISCUSSION TITLE
Stories.json field

### DISCUSSION THREAD

**Discussion by ooihuiying** [storybookjs/storybook#18091] [Help] [answered]
I am working on a project where I will be accessing different projects with storybook installed. For each project, I will run npx sb extract command in their directory to generate stories.json.

I have a separate logic to parse stories.json contents and i want to find the association between the filepath and storyId.
I notice that for different projects, there are different versions of stories.json generated (Eg, v2 or v3). The field importPath does not exist in v2.

Is there anything i can add to the command npx sb extract to force the generated stories.json to contain the field filePath? or ImportPath?

**Comment by shilman** [comment, marked as answer]
My recommendation is to upgrade your projects to 6.4 and use the `buildStoriesJson` feature flag in your `main.js`:

```js
module.exports = {
  features: { buildStoriesJson: true }
}
```

This will automatically build a v3 `stories.json` file when you run `yarn build-storybook` and it will contain the `importPath` field, which contains the information you want.

If you cannot upgrade to 6.4, `npx sb extract` will also generate a `stories.json` file for you. There are a few issues with this approach:
1) It runs `puppeteer` to generate `stories.json`, which is slow.
2) It generates whatever version of `stories.json` corresponds to the version of Storybook you're running
3) In older versions of Storybook, the `filePath` parameter did not get generated in production due to a webpack configuration issue

You can see the fix for it here and manually override the webpack config using in `main.js` using this fix: https://github.com/storybookjs/storybook/commit/631047b370c7439c621229e8e30b0a0cd56d1000

**Reply by ooihuiying** [reply]
Thank you for your response.
